src/handle/why3_tactic.ml

```diff
@@ -61,79 +65,130 @@ let get_config : Sig_state.t -> Pos.popt -> config = fun ss pos ->
+  | _ -> None
+
 (** [translate_term cfg tbl t] translates the given lambdapi term [t] into the
     correspondig Why3 term, using the configuration [cfg] and constants in the
     association list [tbl]. *)
-let translate_term : config -> cnst_table -> term ->
-                       (cnst_table * Why3.Term.term) option = fun cfg tbl t ->
-  let rec translate_prop tbl t =
+let translate_term : config -> cnst_table -> ty_table -> term ->
+                       (cnst_table * ty_table * Why3.Term.term) option = 
+  fun cfg tbl ty_tbl t ->
+  let rec translate_prop tbl ty_tbl t =
     match get_args t with
     | (Symb(s), [t1; t2]) when s == cfg.symb_and ->
-        let (tbl, t1) = translate_prop tbl t1 in
-        let (tbl, t2) = translate_prop tbl t2 in
-        (tbl, Why3.Term.t_and t1 t2)
+        let (tbl, ty_tbl, t1) = translate_prop tbl ty_tbl t1 in
+        let (tbl, ty_tbl, t2) = translate_prop tbl ty_tbl t2 in
+        (tbl, ty_tbl, Why3.Term.t_and t1 t2)
     | (Symb(s), [t1; t2]) when s == cfg.symb_or  ->
-        let (tbl, t1) = translate_prop tbl t1 in
-        let (tbl, t2) = translate_prop tbl t2 in
-        (tbl, Why3.Term.t_or t1 t2)
+        let (tbl, ty_tbl, t1) = translate_prop tbl ty_tbl t1 in
+        let (tbl, ty_tbl, t2) = translate_prop tbl ty_tbl t2 in
+        (tbl, ty_tbl, Why3.Term.t_or t1 t2)
     | (Symb(s), [t1; t2]) when s == cfg.symb_imp ->
-        let (tbl, t1) = translate_prop tbl t1 in
-        let (tbl, t2) = translate_prop tbl t2 in
-        (tbl, Why3.Term.t_implies t1 t2)
+        let (tbl, ty_tbl, t1) = translate_prop tbl ty_tbl t1 in
+        let (tbl, ty_tbl, t2) = translate_prop tbl ty_tbl t2 in
+        (tbl, ty_tbl, Why3.Term.t_implies t1 t2)
     | (Symb(s), [t]     ) when s == cfg.symb_not ->
-        let (tbl, t) = translate_prop tbl t in
-        (tbl, Why3.Term.t_not t)
+        let (tbl, ty_tbl, t) = translate_prop tbl ty_tbl t in
+        (tbl, ty_tbl, Why3.Term.t_not t)
     | (Symb(s), []      ) when s == cfg.symb_bot ->
-        (tbl, Why3.Term.t_false)
+        (tbl, ty_tbl, Why3.Term.t_false)
     | (Symb(s), []      ) when s == cfg.symb_top ->
-        (tbl, Why3.Term.t_true)
+        (tbl, ty_tbl, Why3.Term.t_true)
+    | (Symb(s), [a;Abst(_,t)]   ) when List.memq s [cfg.symb_ex; cfg.symb_all] ->
+        let (ty_tbl, ty) = 
+          match translate_typecode cfg ty_tbl a with 
+          | Some (ty_tbl, ty) -> (ty_tbl, ty) 
+          | None -> assert false  
+        in
+        let x, t = Bindlib.unbind t in
+        let (tbl, ty_tbl ,t) = translate_prop tbl ty_tbl t in
+        let tquant = 
+          let vid = 
+            Why3.(Term.create_vsymbol @@ Ident.id_fresh (Bindlib.name_of x)) ty 
+          in
+          let close = 
+            if s == cfg.symb_ex then Why3.Term.t_exists_close
+            else if s == cfg.symb_all then Why3.Term.t_forall_close
+            else assert false (* Case avoided by pattern matching *)
```

Comment:
can be simplified to `if s == cfg.symb_ex then Why3.Term.t_exists_close else Why3.Term.t_forall_close`